optuna/trial/_trial.py

```diff
@@ -699,13 +699,16 @@ def user_attrs(self) -> Dict[str, Any]:
         return copy.deepcopy(self._cached_frozen_trial.user_attrs)
 
     @property
+    @deprecated_func("3.1.0", "6.0.0")
     def system_attrs(self) -> Dict[str, Any]:
         """Return system attributes.
 
         Returns:
             A dictionary containing all system attributes.
         """
 
+        # TODO(c-bata): Remove the following assertion before merged the PR.
+        assert False, "Must not reach here on unit tests"
```

Comment:
Note, this is just a temporary change to make it easier to detect all places that calls this property method.

```suggestion
```